File: problem3.js
/* Project Euler Largest prime factor
	find largest prime factor of the number 1000 */


function isPrime(number) {
	range = Math.floor(Math.sqrt(number));
	for (var i = range; i > 1; i--) {
		if(number % i == 0) {
			return false;
		}
	}
	return true;
}

function largestPrimeFactor(limit) {
	for (var i = (limit - 1); i > 1; i--) {
		if ((limit % i == 0) && isPrime(i)) {
			return i;
		}
	}
	return "Number started as a prime";
}

console.log(largestPrimeFactor(3));

File: problem3.rb
# Project Euler Largest prime factor


def get_range(num)
	Math.sqrt(num).round
end

def is_prime(number)	
	range = get_range(number)
	i = range
	while i > 1
		if number % i == 0
			return false
		end
		i -= 1
	end
	return true
end

def largestPrimeFactor(number) 
	range = get_range(number)
	primes = []
	(2..range).each do |current|
		if (number % current == 0) && is_prime(current) 
			primes.push(current)
		end
	end
	last = primes.pop
	if last.nil?
		last = "Was prime originally"
	end
	last
end

puts largestPrimeFactor(600851475143)



